File: sample/javascript/text/command_r.js
import { CohereClient } from "cohere-ai";

const cohere = new CohereClient({
  token: process.env.CO_API_KEY,
});

async function main() {
  const response = await cohere.chat({
    chatHistory: [
      { role: "USER", message: "Who discovered gravity?" },
      {
        role: "CHATBOT",
        message:
          "The man who is widely credited with discovering gravity is Sir Isaac Newton",
      },
    ],
    message: "What year was he born?",
    // Web検索を有効にする
    connectors: [{ id: "web-search" }],
  });

  console.log(response.text);
}

main();

File: sample/python/text/command_r.py
import cohere
import os

API_KEY = os.getenv("CO_API_KEY")

co = cohere.Client(API_KEY)

response = co.chat(
    # command-r-plus or command-r
    model="command-r-plus",
    chat_history=[
        {"role": "USER", "message": "日本の首都はどこですか？"},
        {
            "role": "CHATBOT",
            "message": "東京です。",
        },
    ],
    message="では、人口は？",
)

print(response.text)
